### PR TITLE
Added comparison-with-callable message example

### PR DESCRIPTION
**Pull request by matusvalo**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|     | :bug: Bug fix          |
|     | :sparkles: New feature |
|     | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Related #5953

### CODE REVIEW

File: doc/data/messages/c/comparison-with-callable/bad.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+foo = True
+
+
+def bar():
+    pass
+
+
+if foo == bar:  # [comparison-with-callable]
+    pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/comparison-with-callable/bad.py:9]:
```suggestion
fruit: str = "apple"


def function() -> str:
    return "orange"


if fruit == function:  # [comparison-with-callable]
    print("apple == <function function at 0x7f343ff0a1f0>")
```

File: doc/data/messages/c/comparison-with-callable/good.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+foo = True
+
+
+def bar():
+    pass
+
+
+if foo == bar():
+    pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/comparison-with-callable/good.py:9]:
```suggestion
fruit: str = "apple"


def function() -> str:
    return "orange"


if fruit == function():
    print("apple == orange")
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
A proposition because [beginners don't like foo/bar](https://github.com/PyCQA/pylint/pull/6115#issuecomment-1086561103), feel free to modify if you feel creative

**Comment by matusvalo** [comment]
> A proposition because [beginners don't like foo/bar](https://github.com/PyCQA/pylint/pull/6115#issuecomment-1086561103)

OK, I will avoid foo/bar

File: doc/data/messages/c/comparison-with-callable/bad.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+fruit: str = "apple"
+
+
+def function_returning_a_fruit() -> str:
+    return "orange"
+
+
+if fruit == function_returning_a_fruit:  # [comparison-with-callable]
+    print("apple == <function function_returning_a_fruit at 0x7f343ff0a1f0>")
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/comparison-with-callable/bad.py:9]:
```suggestion
def function_returning_a_fruit() -> str:
    return "orange"

def is_an_orange(fruit: str = "apple"):
    # apple == <function function_returning_a_fruit at 0x7f343ff0a1f0>
    return fruit == function_returning_a_fruit  # [comparison-with-callable]
```

File: doc/data/messages/c/comparison-with-callable/good.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+fruit: str = "apple"
+
+
+def function_returning_a_fruit() -> str:
+    return "orange"
+
+
+if fruit == function_returning_a_fruit():
+    print("apple == orange")
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/comparison-with-callable/good.py:9]:
```suggestion
def function_returning_a_fruit() -> str:
    return "orange"

def is_an_orange(fruit: str = "apple"):
    # apple == orange
    return fruit == function_returning_a_fruit()
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
Some possible simplifications I thought off following the other reviews. It's a draft do you plan on adding details/related link ?

**Comment by matusvalo** [comment]
>  It's a draft do you plan on adding details/related link ?

I wanted to review it once again when I will have more time. I think it is fine. Marking as ready for review.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**